src/main/java/org/sagebionetworks/web/server/servlet/JsonLdContentServlet.java

```diff
@@ -0,0 +1,144 @@
+package org.sagebionetworks.web.server.servlet;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.sagebionetworks.client.SynapseClient;
+import org.sagebionetworks.repo.model.Entity;
+import org.sagebionetworks.repo.model.ObjectType;
+import org.sagebionetworks.repo.model.entitybundle.v2.EntityBundle;
+import org.sagebionetworks.repo.model.entitybundle.v2.EntityBundleRequest;
+import org.sagebionetworks.repo.model.table.Dataset;
+import org.sagebionetworks.repo.model.wiki.WikiPage;
+import org.sagebionetworks.web.server.servlet.filter.CrawlFilter;
+import org.sagebionetworks.web.shared.WebConstants;
+
+/**
+ * Given a dataset entity id, will return json in the form: {"jsonld":
+ * "dataset script element content"}
+ *
+ * @author jay
+ *
+ */
+public class JsonLdContentServlet extends HttpServlet {
+
+  private static Logger logger = Logger.getLogger(
+    JsonLdContentServlet.class.getName()
+  );
+  private static final long serialVersionUID = 1L;
+
+  protected static final ThreadLocal<HttpServletRequest> perThreadRequest =
+    new ThreadLocal<HttpServletRequest>();
+
+  private SynapseProvider synapseProvider = new SynapseProviderImpl();
+  private TokenProvider tokenProvider = () ->
+    UserDataProvider.getThreadLocalUserToken(
+      JsonLdContentServlet.perThreadRequest.get()
+    );
+  private RequestHostProvider requestHostProvider = () ->
+    UserDataProvider.getThreadLocalRequestHost(
+      JsonLdContentServlet.perThreadRequest.get()
+    );
+
+  /**
+   * Unit test can override this.
+   */
+  public void setSynapseProvider(SynapseProvider synapseProvider) {
+    this.synapseProvider = synapseProvider;
+  }
+
+  /**
+   * Unit test uses this to provide a mock token provider
+   *
+   * @param tokenProvider
+   */
+  public void setTokenProvider(TokenProvider tokenProvider) {
+    this.tokenProvider = tokenProvider;
+  }
+
+  @Override
+  protected void service(HttpServletRequest arg0, HttpServletResponse arg1)
+    throws ServletException, IOException {
+    JsonLdContentServlet.perThreadRequest.set(arg0);
+    super.service(arg0, arg1);
+  }
+
+  @Override
+  public void service(ServletRequest arg0, ServletResponse arg1)
+    throws ServletException, IOException {
+    super.service(arg0, arg1);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
```

Comment:
Most of this file is boilerplate, with the exception of doGet().  We get the Entity (by ID), and if it's a Dataset then get the Wiki and EntityBundle to construct the Dataset json LD element content. Return this json response